azerothcore/azerothcore-wotlk
# fix(CORE/Quests): Quest dialog fails to update with follow-up quest

## Changes Proposed:
-  Fixes the problem with the quest dialog which doesn't close/update on a follow-up quest if there is no space in the inventory and the quest provides an item.
- New behaviour: If you complete a quest and there is a follow up quest which provides an item, the quest gossip will update so that you can accept the new quest. The "Inventory is full" error will only happen when you try to accept the quest.
-  In case someone uses the auto-accept flag and the circumstances happen, the quest gossip will now get closed.

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6296
- Closes https://github.com/chromiecraft/chromiecraft/issues/827


## Tests Performed:
- Built on Windows 10 x64, Debug and RelWithDebInfo 
- Tested ingame, enUS


## How to Test the Changes:
1. `.go xyz -6 -899 57`
2. `.q add 509` ([quest](https://tbc.wowhead.com/quest=509/elixir-of-agony))
3. Now add as many items so that you only have 3 spaces left in the bag
4. `.additem 3502 7` (at least one more than needed)
5. Now complete the quest at _Apothecary Lydon_
6. The quest gossip should update to the follow-up quest and your inventory should be full
7. No error should be raised during completion of the quest  
8. The "Inventory is full" error should now raise when you try to accept the next quest

Another quest to try this is [Wild Hearts](https://tbc.wowhead.com/quest=429/wild-hearts)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
